### PR TITLE
Roundtrip test and some fixes to inconsistencies between the schema, examples, and JSONMetadataHandler

### PR DESCRIPTION
**Pull request by fedorov**
Followup on #97. 

jsoncompare is from https://github.com/ChannelIQ/jsoncompare, with https://github.com/ChannelIQ/jsoncompare/pull/8 applied.

### CODE REVIEW

File: apps/seg/Testing/CMakeLists.txt [hunk 1]
```diff
@@ -29,5 +32,11 @@ add_test(NAME ${MODULE_NAME}_makeSEG
         --metaDataFileName ${CMAKE_SOURCE_DIR}/doc/seg-example.json
         --segImageFiles ${BASELINE}/liver_seg.nrrd
         --dicomImageFiles ${BASELINE}/01.dcm,${BASELINE}/02.dcm,${BASELINE}/03.dcm
-        --segDICOMFile ${TEMP_DIR}/liver.dcm
+        --segDICOMFile ${MODULE_TEMP_DIR}/liver.dcm
        )
+
+add_test(NAME seg_meta_roundtrip
+         COMMAND python ${CMAKE_SOURCE_DIR}/util/comparejson.py
```
Comment (che85) [inline apps/seg/Testing/CMakeLists.txt]: @fedorov The test is failing, because this python file does not exist. Its name is jsoncompare.py

**Comment by che85** [comment]
@fedorov I suggest, that you first merge my PR after it passed all tests and then I will merge your changes because there will be conflicts....

**Comment by che85** [comment]
@fedorov Actually I can just create a new PR with a combination of those two because I already rebased your branch to mine.

What do you think?

**Comment by fedorov** [comment]
sure, that is fine. I didn't look carefully into why the test fails, because I expected it to fail.

**Comment by che85** [comment]
Why did you expect it to fail?

**Comment by fedorov** [comment]
Because not all of the attributes were populated on read, when I looked at the json converter produces.

Also, comparejson is a separate file I didn't commit. I added it now. I agree naming can be improved to be less confusing.

**Comment by che85** [comment]
@fedorov Can you force push your commits again. Somehow your commits disappeared after I pushed to your github (which actually should not be allowed)

**Comment by fedorov** [comment]
> Can you force push your commits again

done, sorry, overwhelmed by emails... No idea how it was possible for you to alter a repository under my account, you are not even listed as a collaborator under fedorov/dcmqi!

**Pull request closed without merging**